### PR TITLE
Alxngyn/resource model base view

### PR DESCRIPTION
**Pull request by alxngyn**
fixes issue #131 
**needs** PR #212 to be in first

## Changes in this PR.

- [X] Added `resource.html`
- [X] added resource to views.py
- [X] Added resource routing to urls.py
- [X] Changed navigation.html to point to correct resource URL

## Testing this PR.

1. Go to index.html.
2. Click on resource.
3. See a basic html page with links to other things. 
(Some links will be broken but that's on purpose. I wanted to keep this PR simple&incremental and the broken links will be introduced in a new PR)

### Expected Output.

```
$ ./runtests.sh
..................................................................................................................................................................................
----------------------------------------------------------------------
Ran 178 tests in 5.573s

OK
```
fireshot capture 30 - streamwebs - http___localhost_8000_resources_

@osuosl/devs

### CODE REVIEW

File: streamwebs_frontend/streamwebs/templates/streamwebs/resources.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    Here you will find a number of video tutorials covering field protocols for
+    using the Vernier LabQuest and water quality sensors in the field, in addition
+    to a field tutorial for conducting photo point monitoring.
+  </p>
+  <p><a href="glossary">GLOSSARY</a>:
+  </br>
+    The StreamWebs glossary contains some of the most important watershed terms
+    and concepts.
+  </p>
+{% endblock %}
```
Comment (LyonesGamer) [inline streamwebs_frontend/streamwebs/templates/streamwebs/resources.html]: Always make sure you closed all your tags! (You left the div tag at the top unclosed.)

File: streamwebs_frontend/streamwebs/models.py [hunk 2]
```diff
@@ -1064,3 +1064,42 @@ def __str__(self):
...
+        (None, '-----'),
+        ('data_sheet', _('Data Sheet')),
+        ('publication', _('Publication')),
+        ('tutorial_video', _('Tutorial Video')),
+    )
+    name = models.CharField(max_length=255, blank=False)
+    res_type = models.CharField(
+        max_length=255, blank=False, choices=TYPE_CHOICES
+    )
+    # bad way to handle upload_to!  It should be based on res_type
```
Comment (LyonesGamer) [inline streamwebs_frontend/streamwebs/models.py]: Could you clarify what you mean by this? And is it something that should be fixed in this PR?

File: streamwebs_frontend/streamwebs/templates/streamwebs/resources.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+{% block content %}
+<div class="container">
+  <h3>
+    <strong>
+      StreamWebs offers a number of resources to get you started in the field.
+    </strong>
+  </h3>
+  <p><a href="data-sheets">DATA SHEETS</a>:
+  </br>
```
Comment (LyonesGamer) [inline streamwebs_frontend/streamwebs/templates/streamwebs/resources.html]: `br` isn't a closing tag. It should look like either `<br>` or `<br />`. That's a choice of style, but `</br>` is incorrect.

File: streamwebs_frontend/streamwebs/tests/models/test_resource_model.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        #     )
+        for field, field_type in self.expected_fields.items():
+            self.assertEqual(
+                field_type, type(Resource._meta.get_field(field))
+            )
+
+    def test_no_extra_fields(self):
+        # the following is equivalent to MyField._meta.get_all_field_names()
+        # which was deprecated in Django 1.9
+        fields = list(set(chain.from_iterable(
```
Comment (LyonesGamer) [inline streamwebs_frontend/streamwebs/tests/models/test_resource_model.py]: You might want to space this out a bit more with some more newlines and/or comments, this is a bit hard to follow.

File: streamwebs_frontend/streamwebs/models.py (hunk 2 above)
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/models.py]: I think Levi meant to say that the `upload_to={ARGS}` are hard coded and should be dynamic based on the asset type. I think it's fine for now that all "downloadable" objects are in the assets files.

File: streamwebs_frontend/streamwebs/tests/models/test_resource_model.py (hunk 4 above)
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/tests/models/test_resource_model.py]: This code is in all the models. I think @athai found it somewhere.

Comment (leian7) [inline streamwebs_frontend/streamwebs/tests/models/test_resource_model.py]: @LyonesGamer @alxngyn From [here](https://docs.djangoproject.com/en/1.10/ref/models/meta/#migrating-from-the-old-api) I think

**Comment by leian7** [comment]
+1
